bftengine/include/bftengine/MetadataStorage.hpp

```diff
@@ -48,6 +48,9 @@ class MetadataStorage {
   virtual void beginAtomicWriteOnlyBatch() = 0;
   virtual void writeInBatch(uint32_t objectId, char *data, uint32_t dataLength) = 0;
   virtual void commitAtomicWriteOnlyBatch() = 0;
+
+  // In some cases, we would like to load a new metadata after a crash (for example, on reconfiguration actions).
+  virtual void setDontLoadStorageOnStartupFlag() = 0;
```

Comment:
What's the trigger for calling this method, and who does it?

Reply:
The trigger is a reconfiguration action that requires removing the old metadata before shutting down the replicas.
For example, on removing a node, we need to update the node's configuration to commit this node details.
So, this flag will be set, and before the replicas will be shut down, the metadata will be deleted.
Thus, once the replicas reloaded, they will start with the previous state and with no metadata (similar to what @f-squirrel is doing).
The reason why we don't use @f-squirrel tool is that its tool is only for the Merkle tree storage implementation and we want this feature to be part of the reconfiguration feature which is agnostic to the storage implementation.

Reply:
> The reason why we don't use @f-squirrel tool is that its tool is only for the Merkle tree storage implementation and we want this feature to be part of the reconfiguration feature which is agnostic to the storage implementation.
It's doable - just use a proper key manipulator.